libdnf/base/transaction.cpp

```diff
@@ -453,6 +463,21 @@ Transaction::TransactionRunResult Transaction::Impl::run(
 
     db_transaction.set_rpmdb_version_begin(rpm_transaction.get_db_cookie());
     db_transaction.fill_transaction_packages(packages);
+
+    if (!groups.empty()) {
+        // consider currently installed packages + inbound packages as installed for group members
+        rpm::PackageQuery installed_query(base);
+        installed_query.filter_installed();
+        std::set<std::string> installed_names{};
+        for (const auto & pkg : installed_query) {
+            installed_names.emplace(pkg.get_name());
+        }
+        for (const auto & pkg : packages) {
+            installed_names.emplace(pkg.get_package().get_name());
+        }
+        db_transaction.fill_transaction_groups(groups, installed_names);
```

Comment:
I am sorry but the calculation of installed names and using them for transaction groups is somehow strange to me but I am sure it is functional. In goal install and goal_private are kept IDs of packages requested by group to calculate a reason. Also group install can store packages names after calling and verifying of rc==NO_PROBLEM from add_install_to_goal().

Reply:
May be I overlooked something, but how the code handles situation when package is already installed and then the package is installed as a part of group but rpm is not in transaction. I am sure that it will be recorded in db_transaction.fill_transaction_groups() but what about toml for rpms? @lukash Please what's your plans?

Reply:
I think I mostly covered this in my other comment: https://github.com/rpm-software-management/libdnf/pull/1563/commits/1d39537aa4226371e1dedc3ecf6e52e89b152bb0#r932012931 Though I'm not sure we agree on that and it has a number of loose ends.